tests/test_main.py

```diff
@@ -681,30 +685,91 @@ def test_credit_card_number_n(self) -> None:
+        content_provider: FilesProvider = TextProvider([SAMPLES_PATH / "test.html"])
         cred_sweeper = CredSweeper(doc=False)
         cred_sweeper.run(content_provider=content_provider)
         found_credentials = cred_sweeper.credential_manager.get_credentials()
-        assert len(found_credentials) == 0
+        self.assertEqual(0, len(found_credentials))
+
+    # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+
+    def test_data_p(self) -> None:
+
+        def prepare(report: List[Dict[str, Any]]):
+            for x in report:
+                # round ml_probability for macos
+                if x["ml_probability"] is not None:
+                    x["ml_probability"] = round(x["ml_probability"], 5)
+                for y in x["line_data_list"]:
+                    # update windows style path
+                    y["path"] = str(y["path"]).replace('\\', '/')
+                    y["info"] = str(y["info"]).replace('\\', '/')
+                x["line_data_list"].sort(key=lambda k: (
+                    k["path"],
+                    k["line_num"],
+                    k["value"],
+                    k["info"],
+                    k["line"],
+                ))
+            report.sort(key=lambda k: (
+                k["line_data_list"][0]["path"],
+                k["line_data_list"][0]["line_num"],
+                k["line_data_list"][0]["value"],
+                k["line_data_list"][0]["info"],
+                k["line_data_list"][0]["line"],
+                k["rule"],
+                k["severity"],
+                k["ml_probability"],
+            ))
+
+        # do not use parametrised tests with unittests
+        self.maxDiff = 65536
+        # instead the config file is used
+        with open(TESTS_PATH / "data" / ".cfg.json", "r") as f:
+            config = json.load(f)
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            for cfg in config:
+                # important key in .cfg.json is "json_filename"
+                with open(TESTS_PATH / "data" / cfg["json_filename"], "r") as f:
+                    expected_result = json.load(f)
+                prepare(expected_result)
+                tmp_file = Path(tmp_dir) / cfg["json_filename"]
+                # apply the current path to keep equivalence in path
+                os.chdir(TESTS_PATH.parent)
+                content_provider: FilesProvider = TextProvider(["tests/samples"])
+                # replace output report file to place in tmp_dir
+                cfg["json_filename"] = str(tmp_file)
+                cred_sweeper = CredSweeper(**cfg)
+                cred_sweeper.run(content_provider=content_provider)
+                with open(tmp_file, "r") as f:
+                    test_result = json.load(f)
+                prepare(test_result)
+
+                diff = deepdiff.DeepDiff(test_result, expected_result)
+                if diff:
+                    print(json.dumps(test_result, ), flush=True)
```

Comment:
Is it for debug?
If true, please remove this line.

Reply:
It helps to find failure.
When diff present - full output is printed. Then it can be compared or replaced present test data.
Probably, output should write some text what to do...

Reply:
e.g. deepdiff produces this:
```
E               + {'values_changed': {"root[18]['line_data_list'][0]['path']": {'new_value': 'teswts/samples/test.html',
E               +                                                               'old_value': 'tests/samples/test.html'}}} : {'json_filename': '/tmp/tmp81p2kcwc/doc.json', 'doc': True}
```
It is hard to understand what it means and where is the diff.
But if copy and paste print's output - git shows beautiful diff.